cms/djangoapps/contentstore/rest_api/v1/views/tests/test_xblock.py

```diff
@@ -0,0 +1,291 @@
+from unittest.mock import patch
+from django.http import JsonResponse
+
+from django.urls import reverse
+from rest_framework import status
+from rest_framework.test import APITestCase
+from xmodule.modulestore.tests.django_utils import ModuleStoreTestCase
+
+from cms.djangoapps.contentstore.tests.test_utils import AuthorizeStaffTestCase
+
+
+TEST_LOCATOR = "block-v1:dede+aba+weagi+type@problem+block@ba6327f840da49289fb27a9243913478"
+
+
+class XblockViewTestCase(AuthorizeStaffTestCase):
+    # assumes that you want to pass a block id to the url
+
```

Comment:
I recommend adding the following comment here, or something like it

    """
    This base class supports tests with the various HTTP methods (GET, POST, PUT, PATCH, and DELETE).
    Tests for each such message are organized by classes that derive from this one (e.g., XblockViewGetTest).

    Each derived class supplies get_test_data() to govern what goes into the body of the HTTP request.

    Each derived class optionally overrides get_url_params() to govern request parameter values.

    Additionally, each derived class supplies send_request() to bring it all together when making a request
    """